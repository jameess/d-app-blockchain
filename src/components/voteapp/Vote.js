import React, { Component } from 'react';

import Voters from './Voters';
import Owner from './Owner';

import Election from '../../../build/contracts/Election.json';
import getWeb3 from '../..//utils/getWeb3';

class Vote extends Component {
    constructor(props) {
        super(props);
        this.state = {
            web3: null,
            isOwner: false,
        };
    }

    componentWillMount() {
        getWeb3.then(results => {
            this.setState({
                web3: results.web3
            })
            this.renderCorrectPage();
        }).catch(() => {
            console.log('Error finding web3.')
        })
    }

    renderCorrectPage = () => {
        let electionContractInstance;
        const contract = require('truffle-contract');
        const electionContract = contract(Election);
        electionContract.setProvider(this.state.web3.currentProvider);
        this.state.web3.eth.getAccounts((error, accounts) => {
            electionContract.deployed().then((instance) => {
                electionContractInstance = instance
                return electionContractInstance.owner.call();
            }).then((result) => {
                if (result === this.state.web3.eth.accounts[0]) {
                    this.setState(() => ({ isOwner: true }));
                }
                else
                    this.setState(() => ({ isOwner: false }));
            });
        })
    }

    render() {
        return (
            <div>
                <div className="navbar">
                    <ul className="items">
                        <li className="item">
                            <a className="item" href="./home.html">Home</a>
                        </li>
                        <li className="item ">
                            <a href="./home.html#gotoDashboard">Dashboard</a>
                        </li>
                        <li className="item">
                            <a href="./home.html#aboutus">About</a>
                        </li>
                    </ul>
                </div>
                {this.state.isOwner ? <Owner /> : <Voters />}
            </div>
        );
    }
}

export default Vote;
