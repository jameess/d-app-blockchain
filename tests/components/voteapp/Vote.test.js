import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Vote from '../../../src/components/voteapp/Vote';

test('should render NoMatch component', () => {
    const wrapper = shallow(<Vote />);
    expect(toJSON(wrapper)).toMatchSnapshot();
});