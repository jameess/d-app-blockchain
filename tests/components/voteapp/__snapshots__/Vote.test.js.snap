exports[`test should render NoMatch component 1`] = `
<div>
  <div
    className="navbar">
    <ul
      className="items">
      <li
        className="item">
        <a
          className="item"
          href="#">
          Home
        </a>
      </li>
      <li
        className="item ">
        <a
          href="#">
          Dashboard
        </a>
      </li>
      <li
        className="item">
        <a
          href="#">
          About
        </a>
      </li>
    </ul>
  </div>
  <Voters />
</div>
`;
